crates/biome_js_analyze/src/lint/correctness/no_undeclared_variables.rs

```diff
@@ -67,6 +69,20 @@ impl Rule for NoUndeclaredVariables {
                     return None;
                 }
 
+                // arguments object within non-arrow functions
+                if text == "arguments" {
+                    let is_in_non_arrow_function =
+                        identifier.syntax().ancestors().any(|ancestor| {
+                            !matches!(
+                                AnyJsFunction::cast(ancestor),
+                                None | Some(AnyJsFunction::JsArrowFunctionExpression(_))
+                            )
+                        });
+                    if is_in_non_arrow_function {
+                        return None;
+                    }
```

Comment:
I think this check miss some valid undeclared cases like:

```js
const x = arguments;
```

We should seek for a `AnyControlFlowRoot` and check that this node is a function declaration

Reply:
Do you mean `const x = arguments` is a valid case even if it is not in a function?

Reply:
When `const x = arguments` is in the global scope, `arguments` is a regular variable and should be reported as undeclared. Same for other cases like `namespace A { arguments; }` and so on.

Reply:
I think the fix only checks the `arguments` within a function. `arguments` is reported as undeclared in the global scope
https://github.com/biomejs/biome/blob/a49bd8d720c1a6b1fc45e732c1ad7d042ad590be/crates/biome_js_analyze/tests/specs/correctness/noUndeclaredVariables/arguments-object.js.snap?plain=1#L22-L34

Reply:
Thank you! It was an oversight on my part. Sorry.

Reply:
No worries!